jax/abstract_arrays.py

```diff
@@ -16,12 +16,18 @@
 from __future__ import division
 from __future__ import print_function
 
+from collections import Counter
+from itertools import product
+import operator as op
+
 import numpy as onp
 
 from . import core
 from . import ad_util
 from . import dtypes
-from . util import prod, partialmethod
+from . util import prod, partialmethod, safe_map, safe_zip
+map = safe_map
```

Comment:
The code in this file is used so much that it's actually performance-critical; `safe_map`/`safe_zip` are too to slow to use here. Without looking at benchmarks to be sure, I'd say avoid these aliases (i.e. the `map = safe_map` and `zip = safe_zip` lines) and call `safe_map` / `safe_zip` explicitly in this file if/when needed.

Reply:
Ok, done!